cpp/cuproj/CMakeLists.txt

```
@@ -0,0 +1,276 @@
+# causing subsequent runs to have different values for the `Threads::Threads` target.
+# Setting this flag ensures `Threads::Threads` is the same value in first run and subsequent runs.
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+
+# Must come after enable_language(CUDA)
+# Use `-isystem <path>` instead of `-isystem=<path>`
+# because the former works with clangd intellisense
+set(CMAKE_INCLUDE_SYSTEM_FLAG_CUDA "-isystem ")
+
+###################################################################################################
+# - build options ---------------------------------------------------------------------------------
+
+option(BUILD_SHARED_LIBS "Build cuproj shared libraries" ON)
+option(USE_NVTX "Build with NVTX support" ON)
+option(BUILD_TESTS "Configure CMake to build tests" OFF)
+option(BUILD_BENCHMARKS "Configure CMake to build (google) benchmarks" OFF)
+option(PER_THREAD_DEFAULT_STREAM "Build with per-thread default stream" OFF)
+option(DISABLE_DEPRECATION_WARNING "Disable warnings generated from deprecated declarations." OFF)
+# Option to enable line info in CUDA device compilation to allow introspection when profiling / memchecking
+option(CUDA_ENABLE_LINEINFO "Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler" OFF)
+# cudart can be statically linked or dynamically linked. The python ecosystem wants dynamic linking
+option(CUDA_STATIC_RUNTIME "Statically link the CUDA toolkit runtime and libraries" OFF)
+
+message(STATUS "CUPROJ: Build with NVTX support: ${USE_NVTX}")
+message(STATUS "CUPROJ: Configure CMake to build tests: ${BUILD_TESTS}")
+message(STATUS "CUPROJ: Configure CMake to build (google) benchmarks: ${BUILD_BENCHMARKS}")
+message(STATUS "CUPROJ: Build with per-thread default stream: ${PER_THREAD_DEFAULT_STREAM}")
+message(STATUS "CUPROJ: Disable warnings generated from deprecated declarations: ${DISABLE_DEPRECATION_WARNING}")
+message(STATUS "CUPROJ: Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler: ${CUDA_ENABLE_LINEINFO}")
+message(STATUS "CUPROJ: Statically link the CUDA toolkit runtime and libraries: ${CUDA_STATIC_RUNTIME}")
+
+# Set a default build type if none was specified
+rapids_cmake_build_type("Release")
+set(CUPROJ_BUILD_TESTS ${BUILD_TESTS})
+set(CUPROJ_BUILD_BENCHMARKS ${BUILD_BENCHMARKS})
+
+set(CUPROJ_CXX_FLAGS "")
+set(CUPROJ_CUDA_FLAGS "")
+set(CUPROJ_CXX_DEFINITIONS "")
+set(CUPROJ_CUDA_DEFINITIONS "")
+
+# Set RMM logging level
+set(RMM_LOGGING_LEVEL "INFO" CACHE STRING "Choose the logging level.")
+set_property(CACHE RMM_LOGGING_LEVEL PROPERTY STRINGS "TRACE" "DEBUG" "INFO" "WARN" "ERROR" "CRITICAL" "OFF")
+message(STATUS "CUPROJ: RMM_LOGGING_LEVEL = '${RMM_LOGGING_LEVEL}'.")
+
+###################################################################################################
+# - conda environment -----------------------------------------------------------------------------
+
+rapids_cmake_support_conda_env(conda_env MODIFY_PREFIX_PATH)
```

We shouldn't need to modify the prefix path as long as we link to the `conda_env` target.

```suggestion
rapids_cmake_support_conda_env(conda_env)
```

---

Copied from cuSpatial -- perhaps we need to review cuSpatial's cmakelists?